File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    const sheet = {
        "RT": 0,
        "CF": 0,
        "JM": 0,
        "AN": 0
    }
    
    for(let i=0; i<choices.length; i++){
        if(survey[i] in sheet){
            sheet[survey[i]]+=choices[i] - 4;
        }else{
            sheet[survey[i].split("").reverse().join("")]+=4-choices[i];
        }
    }
    if(sheet["RT"] <= 0){
        answer+="R"
    }else{
        answer+="T"
    }
    if(sheet["CF"] <= 0){
        answer+="C"
    }else{
        answer+="F"
    }
    if(sheet["JM"] <= 0){
        answer+="J"
    }else{
        answer+="M"
    }
    if(sheet["AN"] <= 0){
        answer+="A"
    }else{
        answer+="N"
    }
    
    return answer;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;
class Solution {
    public String solution(String[] survey, int[] choices) {
        StringBuilder answer = new StringBuilder();
        Map<String,Integer> map = new LinkedHashMap<>();
        map.put("RT",0);
        map.put("CF",0);
        map.put("JM",0);
        map.put("AN",0);
        
        for(int i=0; i<choices.length; i++){
            if(map.containsKey(survey[i])){
                map.put(survey[i],map.get(survey[i]) + (choices[i]-4));
            }else{
                String reversed = new StringBuilder(survey[i]).reverse().toString();
                map.put(reversed,map.get(reversed)+(4-choices[i]));
            }
        }
        
        for(Map.Entry<String,Integer> entry : map.entrySet()){
            String key = entry.getKey();
            Integer value = entry.getValue();
            
            if(value <= 0){
                answer.append(key.substring(0,1));
            }else{
                answer.append(key.substring(1,2));
            }
        }
        
        return answer.toString();
    }
}